src/dateparser/dateparser.js

```diff
@@ -81,7 +105,7 @@ angular.module('ui.bootstrap.dateparser', [])
       return input;
     }
 
-    format = $locale.DATETIME_FORMATS[format] || format;
+    format = $locale.DATETIME_FORMATS[format] || format.replace(/:/g, '\\:');
```

Comment:
I think a better fix might be to escape all special characters in the regex before doing `new RegExp` on the `createParser` function.

d3 has a nice regexp to do this. See: https://github.com/mbostock/d3/blob/master/src/format/requote.js